libnavigation-core/src/main/java/com/mapbox/navigation/core/trip/MapboxTripStarter.kt

```diff
@@ -0,0 +1,196 @@
+package com.mapbox.navigation.core.trip
+
+import android.annotation.SuppressLint
+import androidx.annotation.VisibleForTesting
+import com.mapbox.android.core.permissions.PermissionsManager
+import com.mapbox.navigation.base.ExperimentalPreviewMapboxNavigationAPI
+import com.mapbox.navigation.core.MapboxNavigation
+import com.mapbox.navigation.core.lifecycle.MapboxNavigationApp
+import com.mapbox.navigation.core.lifecycle.MapboxNavigationObserver
+import com.mapbox.navigation.core.replay.route.ReplayRouteSession
+import com.mapbox.navigation.core.trip.MapboxTripStarterExtra.MAPBOX_TRIP_STARTER_FOLLOW_DEVICE
+import com.mapbox.navigation.core.trip.MapboxTripStarterExtra.MAPBOX_TRIP_STARTER_REPLAY_ROUTE
+import com.mapbox.navigation.core.trip.session.TripSessionState
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.SupervisorJob
+import kotlinx.coroutines.cancel
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.collect
+import kotlinx.coroutines.flow.combine
+import kotlinx.coroutines.flow.update
+import kotlinx.coroutines.launch
+import kotlin.jvm.Throws
+
+/**
+ * This makes it simpler to start a MapboxNavigation trip session. It can be used to enable replay
+ * or to follow the device location. The [MapboxTripStarter] is not able to observe when location
+ * permissions change, so you must notify it of changes. It will check location permissions upon
+ * attach. The location permissions are not required for replay sessions.
+ *
+ * In order to share the state between an App and Android Auto, the instance of this class should
```

Comment:
Maybe sth like "there should be only one instance of this class across the app"? Because "should be shared" does not really say shared between whom.

Reply:
Ok I've took another pass on the wording, please take a look! :)